### ISSUE TITLE
The Ctrl+W shortcut ignores non-Latin words

### ISSUE THREAD

**Issue by Perlence** [anthropics/claude-code#2239]
#### Bug Description

The Ctrl+W shortcut ignores non-Latin words. For example, in the prompt `hello world привет բարև გამარჯობა|`, where "|" is the cursor, pressing Ctrl+W removes too many words, leaving only `hello |`. The Ctrl+W shortcut is a very big part of the muscle memory when working in a terminal.

This issue is made worse by the fact that the Ctrl+Y shortcut doesn't work #1581, so I can't undo the accidental removal of the entire sentence.

For now, I work around this by editing the prompt in a separate editor and pasting the text into Claude Code when I'm done, which is a pretty bad UX.

#### Environment Info

- Platform: darwin
- Terminal: pycharm
- Version: 1.0.25
- Feedback ID: 8a6050de-5edd-4ed5-9e0d-99e9f81a5d0e

**Comment by Perlence** [comment]
This may be a duplicate of #2202.

**Comment by Perlence** [comment]
The Alt+Right and Alt+Left shortcuts are also affected, i.e. they jump to the beginning of the next/previous Latin word.

**Comment by Kolger** [comment]
#1943
